eZ/Publish/Core/Search/Common/Slot/PublishContentTypeDraft.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Search\Common\Slot;
+
+use eZ\Publish\Core\Search\Common\Slot;
+use eZ\Publish\Core\SignalSlot\Signal;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+
+class PublishContentTypeDraft extends Slot
+{
+    /**
+     * Receive the given $signal and react on it.
+     *
+     * @param Signal $signal
+     */
+    public function receive(Signal $signal)
+    {
+        if (!$signal instanceof Signal\ContentTypeService\PublishContentTypeDraftSignal) {
+            return;
+        }
+
+        $contentTypeService = $this->repository->getContentTypeService();
+        // loadContentType, not loadContentTypeDraft, because draft has been already published
+        $contentType = $contentTypeService->loadContentType($signal->contentTypeDraftId);
+        $criterion = new Criterion\ContentTypeId($contentType->id);
+        $query = new Query(['filter' => $criterion]);
+        $results = $this->searchHandler->findContent($query);
+
+        foreach ($results->searchHits as $searchHit) {
+            $contentInfo = $searchHit->valueObject;
+            $this->searchHandler->indexContent($this->persistenceHandler->contentHandler()->load($contentInfo->id, $contentInfo->currentVersionNo));
```

Comment:
Not sure either, but there is also 3rd problem here: indexing should not rely on Search as indexing is by definition asynchronous, so you can't rely on it to return correct data.

This is why we'll need synchronous filtering/search engine for cases like this one. /cc @andrerom